AU2/plugins/util/PoliceRankManager.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+
+from collections import defaultdict
+
+from AU2.database.AssassinsDatabase import ASSASSINS_DATABASE
+from AU2.database.GenericStateDatabase import GENERIC_STATE_DATABASE
+from AU2.database.model import Event, Assassin
+
+DEFAULT_RANKS = {
+    '0': 'Police -2',
+    '1': 'Police -1',
+    '2': 'Police',  # Default police rank
+    '3': 'Police +1',
+    '4': 'Police +2',
+    '5': 'Police +3',
+    '6': 'Police +4',
+    '7': 'Police +5',
+    '8': 'Police +6',
+    '9': 'Umpire',
+}
+DEFAULT_POLICE_RANK = '2'
+
+
+class PoliceRankManager:
+    """
+    Simple manager for police ranks
+    """
+
+    def __init__(self):
+        # from assassin ID to deadline
+        self.assassin_ranks = defaultdict(lambda: DEFAULT_POLICE_RANK)
+        self.activated = GENERIC_STATE_DATABASE.plugin_map.get("PolicePlugin", False)
+
+    def add_event(self, e: Event):
```

Comment:
I think you should stick with the standard of +1 kill = +1 rank to avoid the umpires having to remember. Of those umpires I spoke to, police rank-ups were the most unanimously desired feature to be automated that wasn't already automated in AU1.